### PR TITLE
Address `ast` deprecation warnings

### PR DESCRIPTION
**Pull request by edgarrmondragon**
# Description

* Adapted code for future removal of `ast.Num`, `ast.Str` and `ast.NameConstant`
* Added tests for stable Python 3.12, alpha 3.13, as well as pypy 3.9 and 3.10
* Added `continue-on-error: true` for 2.7 cause it's no longer available with actions/setup-python

# Pre-approval checklist (for submitter)
_Please complete these steps_
- [x] Passes tests
- [ ] New tests for additional features or changed functionality
- [x] My name and contribution added to contributors list (or if I'd rather opt out, I've said so in the PR)

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
The CodeFactor issues seem to be unrelated?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,10 +15,22 @@ jobs:
...
-        python: ['2.7', '3.7', '3.8', '3.9', '3.10', '3.11', '3.12-dev', 'pypy-3.8']
+        python:
+        - '2.7'
+        - '3.7'
+        - '3.8'
+        - '3.9'
+        - '3.10'
+        - '3.11'
+        - '3.12'
+        - '3.13-dev'
```
Comment (kurtmckee) [inline .github/workflows/ci.yml:30]: In a future iteration, I recommend changing this to simply `'3.13'` and adding [the `allow-prereleases` flag](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases) to the `setup-python` action.

Comment (kurtmckee) [inline .github/workflows/ci.yml:30]: In a future iteration, I recommend changing this to simply `'3.13'` and adding [the `allow-prereleases` flag](https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases) to the `setup-python` action.

**Review by kurtmckee — APPROVED** [review]
Looks good!

@danthedeckie Please review this PR and merge -- it should resolve newer Python `ast` warnings.

**Comment by edgarrmondragon** [comment]
All tests (except for Python 2.7) are 🟢 over on my fork: https://github.com/edgarrmondragon/simpleeval/actions/runs/9782703954

Screenshot 2024-07-03 at 12 04 56 p m

File: .github/workflows/ci.yml (hunk 1 above)
Comment (edgarrmondragon) [reply .github/workflows/ci.yml]: f4b72ede1e109c3c341783d11816dd33d35b432c

**Comment by edgarrmondragon** [comment]
Ping @danthedeckie

**Comment by danthedeckie** [comment]
Thanks @edgarrmondragon !

I've merged this via #146 , using a slightly different method now that it makes sense to drop old old pythons.

Thank you for helping push this all forward.

**Comment by edgarrmondragon** [comment]
Thank you @danthedeckie!

**Pull request closed without merging**